jupyterhub/tests/selenium/test_browser.py

```diff
@@ -0,0 +1,228 @@
+import asyncio
+import time
+from functools import partial
+
+import pytest
+from selenium import webdriver
+from selenium.common.exceptions import NoSuchElementException, TimeoutException
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+from tornado.escape import url_escape
+from tornado.httputil import url_concat
+
+from jupyterhub.tests.selenium.locators import (
+    HomePageLocators,
+    LoginPageLocators,
+    TokenPageLocators,
+)
+from jupyterhub.utils import exponential_backoff
+
+from ...utils import url_path_join
+from ..utils import public_host, public_url, ujoin
+
+pytestmark = pytest.mark.selenium
+
+
+async def webdriver_wait(driver, condition, timeout=30):
```

Comment:
Docstring for wait - explain that it's an async wrapper for selenium's wait function, and that `condition` is something from selenium's expected_conditions